public/app/features/alerting/unified/api/buildInfo.ts

```diff
@@ -85,24 +95,42 @@ export async function discoverDataSourceFeatures(dsSettings: {
     };
   }
 
-  // if no features are reported but buildinfo was returned we're talking to Prometheus
+  // Otherwise if we don't have build info, but we did trigger the fallback, we should be dealing with a prometheus datasource that's older than 2.14.0
+  else if (!hasBuildInfo) {
+    return {
+      version: metric?.version,
+      application: PromApplication.Prometheus,
+      features: {
+        rulerApiEnabled: false,
+      },
+    };
+  }
+
+  // if no features are reported but buildinfo was returned we're talking to modern Prometheus
   const { features } = buildInfoResponse.data;
   if (!features) {
     return {
+      version: buildInfoResponse.data.version,
       application: PromApplication.Prometheus,
       features: {
         rulerApiEnabled: false,
       },
     };
   }
 
-  // if we have both features and buildinfo reported we're talking to Mimir
-  return {
-    application: PromApplication.Mimir,
-    features: {
-      rulerApiEnabled: features?.ruler_config_api === 'true',
-    },
-  };
+  if (buildInfoResponse.data.application === 'Grafana Mimir') {
+    // Mimir does return a version, but this is the mimir version, and not the prometheus version.
+    // We do not want to use mimir version numbers to detect feature support, we should instead ask the mimir team to add new flags.
+    return {
```

Comment:
Or should we add the mimir version so we can display the result in the test data source UI? Then we need to make sure developers don't try to use that version number anywhere...

Reply:
I didn't want people to confuse the mimir version for the prometheus version, and since we don't currently have any plan to use the mimir version because they are providing feature flags from the buildinfo api endpoint, and they have asked us to work with them to add new feature flags instead of using the version number.

But if we're using it for analytics or just to show in the UI that's fine.